### PR TITLE
Optimize ContractManifest.Clone()

### PR DESCRIPTION
**Pull request by erikzhang**
Close https://github.com/neo-project/neo/pull/1326

### CODE REVIEW

File: src/neo/SmartContract/Manifest/ContractPermission.cs [hunk 1]
```diff
@@ -27,6 +27,15 @@ public class ContractPermission
             Methods = WildCardContainer<string>.CreateWildcard()
         };
 
+        public ContractPermission Clone()
+        {
+            return new ContractPermission
+            {
+                Contract = Contract,
+                Methods = Methods
```
Comment (shargon) [inline src/neo/SmartContract/Manifest/ContractPermission.cs]: Also, we need to clone this

Comment (erikzhang) [reply src/neo/SmartContract/Manifest/ContractPermission.cs]: `Methods` is immutable.

**Comment by Tommo-L** [comment]
Waiting for minutes, we are testing the performance

**Comment by eryeer** [comment]
After Performance test, these change can get a equivalent improvement to the pr #1326.

**Pull request merged**